### ISSUE TITLE
Unable to change license owner or add users

### ISSUE THREAD

**Issue by MaxWaldorf** [keygen-sh/keygen-api#920]
I'm trying for the first time the feature added in 1.3.0 but anytime I'm trying to change the owner of a license, I'm stuck with an error...

I am using an "admin-token" to perform the task...

Here is the query:
```
curl -X 'PUT' \
  'https://REDACTED/v1/accounts/REDACTED/licenses/REDACTED/owner' \
  -H 'accept: application/vnd.api+json' \
  -H 'Authorization: Bearer admin-REDACTED' \
  -H 'Content-Type: application/vnd.api+json' \
  -d '{
  "data": {
    "type": "users",
    "id": "REDACTED"
  }
}'
```

Error:

```
{
  "meta": {
    "id": "REDACTED"
  },
  "errors": [
    {
      "title": "Access denied",
      "detail": "You do not have permission to complete the request (admin lacks permission to perform action)"
    }
  ]
}
```

**Comment by MaxWaldorf** [comment]
I did the migration in the container here is the result:

```
keygen-658dbc9bd-nlxzz:~$ rake db:migrate
keygen-658dbc9bd-nlxzz:~$ rake db:seed
keygen-658dbc9bd-nlxzz:~$ rake keygen:permissions:admins:add[license.users.attach,license.users.detach,license.owner.update,machine.owner.update]
rake keygen:permissions:environments:add[license.users.attach,license.users.detach,license.owner.update,machine.owner.update]
rake keygen:permissions:products:add[license.users.attach,license.users.detach,license.owner.update,machine.owner.update]
I, [2024-12-23T00:33:55.283004 #124]  INFO -- : Adding license.users.attach,license.users.detach,license.owner.update,machine.owner.update permissions to 1 admins...
W, [2024-12-23T00:33:55.284505 #124]  WARN -- : Scoped order is ignored, it's forced to be batch order.
W, [2024-12-23T00:33:55.310597 #124]  WARN -- : Scoped order is ignored, it's forced to be batch order.
I, [2024-12-23T00:33:55.857012 #124]  INFO -- : Skipping REDACTED_ADMIN_USER..
I, [2024-12-23T00:33:55.857217 #124]  INFO -- : Done
I, [2024-12-23T00:34:12.760031 #130]  INFO -- : Adding license.users.attach,license.users.detach,license.owner.update,machine.owner.update permissions to 0 environments...
W, [2024-12-23T00:34:12.760907 #130]  WARN -- : Scoped order is ignored, it's forced to be batch order.
I, [2024-12-23T00:34:12.784648 #130]  INFO -- : Done
I, [2024-12-23T00:34:29.263786 #136]  INFO -- : Adding license.users.attach,license.users.detach,license.owner.update,machine.owner.update permissions to 1 products...
W, [2024-12-23T00:34:29.264774 #136]  WARN -- : Scoped order is ignored, it's forced to be batch order.
W, [2024-12-23T00:34:29.288943 #136]  WARN -- : Scoped order is ignored, it's forced to be batch order.
I, [2024-12-23T00:34:29.842273 #136]  INFO -- : Skipping REDACTED_ONLY_PRODUCT...
I, [2024-12-23T00:34:29.842440 #136]  INFO -- : Done
```

By the way, the rake version is not good in the container version...

**Comment by ezekg** [comment]
What version are you upgrading from? I'd wager to bet that your admin's permissions don't have the permissions introduced in v1.1, so the admin doesn't have the "default" permission set anymore and is being treated as if having one-off permissions and is being skipped (that's what it means when it says "`Skipping REDACTED_ADMIN_USER...`").

The missing permissions from v1.1 are [here](https://github.com/keygen-sh/keygen-api/pull/679). I'll admit this is poorly documented, so I'll try and update the v1.3 upgrade instructions, and add instructions for v1.1 as well. Not sure how that was missed.

Try running this instead, to catch up from v1.1 onwards:

```bash
rake keygen:permissions:admins:add[package.create,package.delete,package.read,package.update,engine.read,license.users.attach,license.users.detach,license.owner.update,machine.owner.update]
rake keygen:permissions:environments:add[package.create,package.delete,package.read,package.update,engine.read,license.users.attach,license.users.detach,license.owner.update,machine.owner.update]
rake keygen:permissions:products:add[package.create,package.delete,package.read,package.update,engine.read,license.users.attach,license.users.detach,license.owner.update,machine.owner.update]
rake keygen:permissions:licenses:add[package.read,engine.read]
rake keygen:permissions:users:add[package.read,engine.read]
```

If that doesn't work, let me know what the output of this is from a console:

```rb
Permission::ADMIN_PERMISSIONS - Account.last.admins.last.permissions.actions
```

Sorry for the trouble! Need to work on improving the upgrade DX.

**Comment by MaxWaldorf** [comment]
So yes, I upgraded from several versions 1.1, 1.2 then 1.3
But things don't go great by doing this...

None of the permissions are truely added, I have to do it manually from DB.

Is it because I'm using budle exec?
Seems it is reading DB properly but injecting properties doesn't seem to succeed although it says it has...

I'm not really familliar with console, I'm mostly using an updated swagger API portal...

**Comment by ezekg** [comment]
What do you mean exactly by "things don't go great by doing this?" Everything looks like it's running as expected — no errors — you're simply missing a few permissions for it to succeed.

Did you run what I outlined in my last post?

**Comment by MaxWaldorf** [comment]
I did run what you recommended but it did not add the permission to the users or licenses.

If I check the permission ID on the licenses or users than they do not show up in the SQL query in postgresql.

I eventually for product and admin to add missing permissions via SQL.

I did add a * permission for admin user.

I am wondering if there is a debug mode to see the queries going through maybe.

I'll to see what I can provide you to debug if needed.

Thanks for your time
